nsqd/protocol_v2.go

```diff
@@ -124,7 +124,9 @@ func (p *protocolV2) IOLoop(conn net.Conn) error {
 
 func (p *protocolV2) SendMessage(client *clientV2, msg *Message) error {
 	p.ctx.nsqd.logf(LOG_DEBUG, "PROTOCOL(V2): writing msg(%s) to client(%s) - %s", msg.ID, client, msg.Body)
-	var buf = &bytes.Buffer{}
+
+	buf := bufferPoolGet()
+	defer bufferPoolPut(buf)
```

Comment:
looks like defers finally got basically as fast as inline calls, which means we can separately clean up a lot of places we were avoiding them...

Reply:
Yes, but so what do you want me to modify here?

Reply:
no, I think nothing to modify here, it was just a side comment that we used to avoid using defer in performance-sensitive functions, but now it is fine